tests/suite/resources_utils.py

```diff
@@ -1174,10 +1174,21 @@ def ensure_response_from_backend(req_url, host, additional_headers=None) -> None
     headers = {"host": host}
     if additional_headers:
         headers.update(additional_headers)
-    for _ in range(30):
-        resp = requests.get(req_url, headers=headers, verify=False)
-        if resp.status_code != 502 and resp.status_code != 504:
-            print(f"After {_ * 2} seconds got non 502|504 response. Continue with tests...")
-            return
-        time.sleep(1)
-    pytest.fail(f"Keep getting 502|504 from {req_url} after 60 seconds. Exiting...")
+    
+    if check404:
```

Comment:
Woudl it make sense to put the status checks in a function and thus remove if...else completely? something like that

```
for _ in range(60):
            resp = requests.get(req_url, headers=headers, verify=False)
            if  checkRespStatus(check404):
                print(f"After {_} retries at 1 second interval, got {resp.status_code} response. Continue with tests...")
                return
            time.sleep(1)
        pytest.fail(f"Keep getting {resp.status_code} from {req_url} after 60 seconds. Exiting...")
```